torch/ao/quantization/backend_config/README.md

```diff
@@ -17,4 +41,110 @@ tensor_1            tensor_2
       nn.ReLU
 ```
 
-we’ll match the last node as the anchor point of the match, and we can retrieve the whole graph by tracing back from the node, e.g. in the example above, we matched nn.ReLU node, then node.args[0] is the operator.add node.
+During prepare and convert, we’ll match the last node, which will be the anchor point of the match, and we can retrieve the whole graph by tracing back from the node. E.g. in the example above, we matched the `nn.ReLU` node, and `node.args[0]` is the `operator.add` node.
+
+## BackendConfig Specification
+
+The BackendConfig is comprised of a list of BackendPatternConfigs, each of which define the specifications and the requirements for an operator pattern. Here is an example usage:
+
+```
+import torch
+from torch.ao.quantization.backend_config import BackendConfig, BackendPatternConfig, DTypeConfig, ObservationType
+from torch.ao.quantization.fuser_method_mappings import reverse_sequential_wrapper2
+
+weighted_int8_dtype_config = DTypeConfig(
+    input_dtype=torch.quint8,
+    output_dtype=torch.quint8,
+    weight_dtype=torch.qint8,
+    bias_type=torch.float)
+
+linear_config = BackendPatternConfig(torch.nn.Linear) \
+    .set_observation_type(ObservationType.OUTPUT_USE_DIFFERENT_OBSERVER_AS_INPUT) \
+    .add_dtype_config(weighted_int8_dtype_config) \
+    .set_root_module(torch.nn.Linear) \
+    .set_qat_module(torch.ao.nn.qat.Linear) \
+    .set_reference_quantized_module(torch.ao.nn.quantized.reference.Linear)
+
+conv_relu_config = BackendPatternConfig((torch.nn.ReLU, torch.nn.Conv2d)) \
+    .set_observation_type(ObservationType.OUTPUT_USE_DIFFERENT_OBSERVER_AS_INPUT) \
+    .add_dtype_config(weighted_int8_dtype_config) \
+    .set_fused_module(torch.ao.nn.intrinsic.ConvReLU2d) \
+    .set_fuser_method(reverse_sequential_wrapper2(torch.ao.nn.intrinsic.ConvReLU2d))
+
+backend_config = BackendConfig("my_backend") \
+    .set_backend_pattern_config(linear_config) \
+    .set_backend_pattern_config(conv_relu_config)
+```
+#### set_observation_type
```

Comment:
I feel some of these should be moved to docstrings for the API?

Reply:
I think we should discuss about the goal for the README, my current understanding is that the README should describe the design decisions and how it is implemented

Reply:
I think a key thing would be to avoid duplication.  Either we should have these examples in docstrings or here, but not in both places.  If it's duplicated the risk of it being outdated increases.

Reply:
I do feel like what is in this file now would fit more naturally in documentation, this seems to be documentation for users of BackendConfig, not necessarily developers of AO code.